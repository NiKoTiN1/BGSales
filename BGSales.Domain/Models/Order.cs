using System;

namespace BGSales.Domain.Models
{
    public class Order : IEntity
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public int AudienceAge { get; set; }
        public string Description { get; set; }
        public Businessman Advertiser { get; set; }
        public string AdvertiserId { get; set; }
        public double Budget { get; set; }
        public DateTime CreateDate { get; set; }
    }
}
