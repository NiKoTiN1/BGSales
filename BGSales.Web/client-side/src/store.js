import { createStore, combineReducers, applyMiddleware } from "redux";
import reducer from "./reducers";
import thunk from "redux-thunk";
var rootReducer = combineReducers({
    reducer: reducer,
});
var store = createStore(rootReducer, applyMiddleware(thunk));
export default store;
