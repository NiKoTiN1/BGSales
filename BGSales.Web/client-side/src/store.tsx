import { createStore, combineReducers, applyMiddleware } from "redux";
import reducer from "./reducers";
import thunk from "redux-thunk";

const rootReducer = combineReducers({
  reducer,
});

const store = createStore(rootReducer, applyMiddleware(thunk));

export default store;
