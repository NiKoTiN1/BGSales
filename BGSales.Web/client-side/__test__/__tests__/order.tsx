import { Provider } from "react-redux";
import renderer from 'react-test-renderer';
import Order from "../../src/components/order";
import CreateOrder from "../../src/components/create-order";
import OrderEdit from "../../src/components/order-edit";
import Orders from "../../src/components/orders";
import store from "../../src/store"
import React from 'react';

jest.mock("history");
jest.useFakeTimers()
describe("BGSales Order module", () => {
	const history: any = 0;
	it("check be truth base element footer ", () => {

		const wrapper = renderer.create(<Provider store={store}><Order id={""} /> </Provider>);
		expect(wrapper.toJSON()).toMatchSnapshot();
	});


	it("check be truth create order ", () => {

		const wrapper = renderer.create(<Provider store={store}><CreateOrder history={history}  /> </Provider>);
		expect(wrapper.toJSON()).toMatchSnapshot();
	});

	it("check be truth order edit ", () => {

		const wrapper = renderer.create(<Provider store={store}><OrderEdit history={history} /> </Provider>);
		expect(wrapper.toJSON()).toMatchSnapshot();
	});


	it("check be truth orders", () => {

		const wrapper = renderer.create(<Provider store={store}><Orders history={history} /> </Provider>);
		expect(wrapper.toJSON()).toMatchSnapshot();
	});
});