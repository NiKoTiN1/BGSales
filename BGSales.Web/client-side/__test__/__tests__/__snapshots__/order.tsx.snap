// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BGSales Order module check be truth base element footer  1`] = `
Array [
  <div
    className="container-error"
  >
    <img
      src={
        Object {
          "process": [Function],
        }
      }
    />
    <p>
      Error this page is not available
    </p>
  </div>,
  " ",
]
`;

exports[`BGSales Order module check be truth base element user menu  1`] = `
Array [
  <div
    className="container-error"
  >
    <img
      src={
        Object {
          "process": [Function],
        }
      }
    />
    <p>
      Error this page is not available
    </p>
  </div>,
  " ",
]
`;

exports[`BGSales Order module check be truth create order  1`] = `
Array [
  <div
    className="container-error"
  >
    <img
      src={
        Object {
          "process": [Function],
        }
      }
    />
    <p>
      Error this page is not available
    </p>
  </div>,
  " ",
]
`;

exports[`BGSales Order module check be truth order edit  1`] = `
Array [
  <div
    className="container-error"
  >
    <img
      src={
        Object {
          "process": [Function],
        }
      }
    />
    <p>
      Error this page is not available
    </p>
  </div>,
  " ",
]
`;

exports[`BGSales Order module check be truth orders 1`] = `
Array [
  <div
    className="container-error"
  >
    <img
      src={
        Object {
          "process": [Function],
        }
      }
    />
    <p>
      Error this page is not available
    </p>
  </div>,
  " ",
]
`;
